_studio/mfx_lib/decode/vp8/src/mfx_vp8_dec_decode_hw.cpp

```diff
@@ -1003,309 +1003,312 @@ mfxStatus VideoDECODEVP8_HW::DecodeFrameHeader(mfxBitstream *in)
+        }
 
-    mfxU8 bits = (uint8_t)m_boolDecoder[VP8_FIRST_PARTITION].decode(7);
+        mfxU8 bits = (uint8_t)m_boolDecoder[VP8_FIRST_PARTITION].decode(7);
 
-    m_frame_info.loopFilterType  = bits >> 6;
-    m_frame_info.loopFilterLevel = bits & 0x3F;
+        m_frame_info.loopFilterType  = bits >> 6;
+        m_frame_info.loopFilterLevel = bits & 0x3F;
 
-    bits = (uint8_t)m_boolDecoder[VP8_FIRST_PARTITION].decode(4);
+        bits = (uint8_t)m_boolDecoder[VP8_FIRST_PARTITION].decode(4);
 
-    m_frame_info.sharpnessLevel     = bits >> 1;
-    m_frame_info.mbLoopFilterAdjust = bits & 1;
+        m_frame_info.sharpnessLevel     = bits >> 1;
+        m_frame_info.mbLoopFilterAdjust = bits & 1;
 
-    m_frame_info.modeRefLoopFilterDeltaUpdate = 0;
+        m_frame_info.modeRefLoopFilterDeltaUpdate = 0;
 
-    if (m_frame_info.mbLoopFilterAdjust)
-    {
-        m_frame_info.modeRefLoopFilterDeltaUpdate = (uint8_t)m_boolDecoder[VP8Defs::VP8_FIRST_PARTITION].decode();
-        if (m_frame_info.modeRefLoopFilterDeltaUpdate)
+        if (m_frame_info.mbLoopFilterAdjust)
         {
-            UpdateLoopFilterDeltas(m_boolDecoder[VP8_FIRST_PARTITION]);
+            m_frame_info.modeRefLoopFilterDeltaUpdate = (uint8_t)m_boolDecoder[VP8Defs::VP8_FIRST_PARTITION].decode();
+            if (m_frame_info.modeRefLoopFilterDeltaUpdate)
+            {
+                UpdateLoopFilterDeltas(m_boolDecoder[VP8_FIRST_PARTITION]);
+            }
         }
-    }
-
-    mfxU32 partitions;
 
-    bits = (uint8_t)m_boolDecoder[VP8_FIRST_PARTITION].decode(2);
+        mfxU32 partitions;
 
-    m_CodedCoeffTokenPartition = bits;
+        bits = (uint8_t)m_boolDecoder[VP8_FIRST_PARTITION].decode(2);
 
-    partitions = 1 << bits;
-    m_frame_info.numTokenPartitions = 1 << bits;
+        m_CodedCoeffTokenPartition = bits;
 
-    m_frame_info.numPartitions = m_frame_info.numTokenPartitions;
-    partitions =  m_frame_info.numPartitions;
-    mfxU8 *pTokenPartition = data_in + first_partition_size;
+        partitions = 1 << bits;
+        m_frame_info.numTokenPartitions = 1 << bits;
 
-    if (partitions > 1)
-    {
-        m_frame_info.partitionStart[0] = pTokenPartition + (partitions - 1) * 3;
+        m_frame_info.numPartitions = m_frame_info.numTokenPartitions;
+        partitions =  m_frame_info.numPartitions;
+        mfxU8 *pTokenPartition = data_in + first_partition_size;
 
-        for (uint32_t i = 0; i < partitions - 1; i++)
+        if (partitions > 1)
         {
-            m_frame_info.partitionSize[i] = (int32_t)(pTokenPartition[0]) |
-                                             (pTokenPartition[1] << 8) |
-                                             (pTokenPartition[2] << 16);
-            pTokenPartition += 3;
+            m_frame_info.partitionStart[0] = pTokenPartition + (partitions - 1) * 3;
+
+            if (m_frame_info.partitionStart[0] > data_in_end)
+                MFX_RETURN(MFX_ERR_MORE_DATA);
```

Comment:
Out of bounds when reading pTokenPartion data from bitstream data.
Add checking of remaining bitstream data size.